cpu/efm32_common/include/periph_cpu.h

```diff
@@ -0,0 +1,339 @@
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_efm32_common
+ * @{
+ *
+ * @file
+ * @brief       CPU specific definitions for internal peripheral handling
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ */
+
+#ifndef PERIPH_CPU_H
+#define PERIPH_CPU_H
+
+#include "mutex.h"
+
+#include "cpu_conf.h"
+
+#include "em_adc.h"
+#include "em_cmu.h"
+#include "em_device.h"
+#include "em_gpio.h"
+#include "em_usart.h"
+#ifdef _SILICON_LABS_32B_PLATFORM_1
+#include "em_dac.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Enable support for Low-power peripherals (if supported by CPU).
+ * @{
+ */
+#ifndef LOW_POWER_ENABLED
+#define LOW_POWER_ENABLED   (1)
+#endif
+/** @} */
+
+/**
+ * @brief   Internal macro for combining ADC resolution (x) with number of
+ *          shifts (y).
+ */
+#define ADC_MODE(x, y)     ((y << 4) | x)
+
+/**
+ * @brief   Possible ADC resolution settings
+ * @{
+ */
+#define HAVE_ADC_RES_T
+typedef enum {
+    ADC_RES_6BIT = ADC_MODE(adcRes6Bit, 0),     /**< ADC resolution: 6 bit */
+    ADC_RES_8BIT = ADC_MODE(adcRes8Bit, 0),     /**< ADC resolution: 8 bit */
+    ADC_RES_10BIT = ADC_MODE(adcRes12Bit, 2),   /**< ADC resolution: 10 bit (shifted from 12 bit) */
+    ADC_RES_12BIT = ADC_MODE(adcRes12Bit, 0),   /**< ADC resolution: 12 bit */
+} adc_res_t;
+/** @} */
+
+/**
+ * @brief   ADC device configuration
+ * @{
+ */
+typedef struct {
+    ADC_TypeDef *dev;                 /**< ADC device used */
+    CMU_Clock_TypeDef cmu;            /**< the device CMU channel */
+} adc_conf_t;
+
+typedef struct {
+    uint8_t dev;                      /**< device index */
+#ifdef _SILICON_LABS_32B_PLATFORM_1
+    ADC_SingleInput_TypeDef input;    /**< input channel */
+#else
+    ADC_PosSel_TypeDef input;         /**< input channel */
+#endif
+    ADC_Ref_TypeDef reference;        /**< channel voltage reference */
+    ADC_AcqTime_TypeDef acq_time;     /**< channel acquisition time */
+} adc_chan_conf_t;
+/** @} */
+
+/**
+ * @brief   Length of CPU ID in octets.
+ */
+#define CPUID_LEN           (8U)
+
+/**
+ * @brief   DAC device configuration
+ * @{
+ */
+#if defined(DAC_COUNT) && DAC_COUNT > 0
+typedef struct {
+    DAC_TypeDef *dev;       /**< DAC device used */
+    CMU_Clock_TypeDef cmu;  /**< the device CMU channel */
+} dac_conf_t;
+
+typedef struct {
+    uint8_t dev;            /**< device index */
+    uint8_t index;          /**< channel index */
+    DAC_Ref_TypeDef ref;    /**< channel voltage reference */
+} dac_chan_conf_t;
+#endif
+/** @} */
+
+/**
+ * @brief   Define a custom type for GPIO pins.
+ * @{
+ */
+#define HAVE_GPIO_T
+typedef uint32_t gpio_t;
+/** @} */
+
+/**
+ * @brief   Definition of a fitting UNDEF value.
+ */
+#define GPIO_UNDEF          (0xffffffff)
```

Comment:
IMHO no need to redefine this, as the default definition of this value is the same...

Reply:
It seems that is necessary to redefine this value, because `periph_conf.h` depends on `GPIO_UNDEF`, for which I need to import `periph/gpio.h`, which depends on `periph_conf.h` again. 

(This isn't the case for the SLTB001a, but it is for the STK3600)

The same applies when I want to use the default type for `gpio_t`. For instance in `periph_cpu.c` I define `spi_dev_t` to use `gpio_t`, which is not defined unless I include `periph/gpio.h`,  which depends on `periph_cpu.h` again.

Any advice?

Reply:
I see, or more precise: I remember. Have been running into the same problem for other CPUs. So for this case, just leave it as is.

I have actually started to draft out a solution to this problem, re-structuring the CPU includes: #6999. But had not a chance yet to proceed with this.